dotCMS/src/main/java/com/dotcms/rest/api/v1/vtl/VTLResource.java

```diff
@@ -60,86 +64,127 @@ public VTLResource() {
-        final MultivaluedMap<String, String> queryParameters = uriInfo.getQueryParameters();
-
-        final DotJSONCacheKey dotJSONCacheKey = getDotJSONCacheKey(request, initDataObject);
 
-        final Optional<DotJSON> dotJSONOptional = CacheLocator.getDotJSONCache()
-                .get(dotJSONCacheKey);
+        final DotJSONCacheStrategy cacheStrategy = getCacheStrategy(HTTPMethod.GET);
+        final Optional<DotJSON> dotJSONOptional = cacheStrategy.get(request, initDataObject.getUser());
 
         if(dotJSONOptional.isPresent()) {
             return Response.ok(dotJSONOptional.get()).build();
         }
 
-        DotJSON<String, String> dotJSON;
-
         try {
-            final Host site = this.hostAPI.resolveHostName(request.getServerName(), APILocator.systemUser(), false);
-            final String getFilePath = VTL_PATH + StringPool.SLASH + folderName + StringPool.SLASH
-                    + HTTPMethod.GET.fileName + FILE_EXTENSION;
-            final Identifier identifier = identifierAPI.find(site, getFilePath);
-            final Contentlet getFileContent = contentletAPI.findContentletByIdentifier(identifier.getId(), true,
-                    currentLanguage.getId(), user, true);
-            final FileAsset getFileAsset = APILocator.getFileAssetAPI().fromContentlet(getFileContent);
-
-            final org.apache.velocity.context.Context context = VelocityUtil.getInstance().getContext(request, response);
-            context.put("urlParams", initDataObject.getParamsMap());
-            context.put("queryParams", queryParameters);
-            context.put("dotJSON", new DotJSON());
-
-            final StringWriter evalResult = new StringWriter();
-
-            try (final InputStream fileAssetIputStream = getFileAsset.getInputStream()) {
-                VelocityUtil.getEngine().evaluate(context, evalResult, "", fileAssetIputStream);
-                dotJSON = (DotJSON<String, String>) context.get("dotJSON");
+            final FileAsset getFileAsset = getVTLFile(HTTPMethod.GET, request, folderName, initDataObject.getUser());
 
-                if(dotJSON.size()==0) { // If dotJSON is not used let's return the raw evaluation of the velocity file
-                    return Response.ok(evalResult.toString()).build();
-                }
+            final Map<String, Object> contextParams = CollectionsUtils.map(
+                    "urlParams", initDataObject.getParamsMap(),
+                    "queryParams", uriInfo.getQueryParameters());
 
-                // let's add it to cache
-                CacheLocator.getDotJSONCache().add(dotJSONCacheKey, dotJSON);
-            }
+            return evalVTLFile(request, response, getFileAsset, contextParams,
+                    initDataObject.getUser(), cacheStrategy);
         } catch(DotContentletStateException e) {
             final String errorMessage = "Unable to find velocity file '" + HTTPMethod.GET.fileName + FILE_EXTENSION
```

Comment:
use a StringBuilder instead

Reply:
After replacing with StringBuilder I got this warning from IntelliJ: 

>StringBuilder' can be replaced with 'String'
Reports any usages of java.lang.StringBuffer and java.lang.StringBuilder which can be replaced with a single java.lang.String concatenation. Using a String concatenation makes the code shorter and simpler. This inspection only reports when the resulting concatenation is at least as efficient or more efficient than the original StringBuffer or StringBuilder code.

So leaving as it was